integration-tests/src/test/java/org/apache/fineract/integrationtests/SchedulerJobsTestResults.java

```diff
@@ -339,33 +339,57 @@ public void testApplyHolidaysToLoansJobOutcome() throws InterruptedException {
-
-        for (LinkedHashMap period : periods) {
-            ArrayList<Integer> fromDate = (ArrayList<Integer>) period.get("fromDate");
-            if (fromDate != null && Objects.equals(fromDate.get(1), repaymentsRescheduledDate.get(1))) {
-                Assertions.assertNotEquals(repaymentsRescheduledDate.get(2), fromDate.get(2),
-                        "Verifying Repayment Rescheduled Day before Running Apply Holidays to Loans Scheduler Job");
+        final LinkedHashMap repaymentScheduleHashMapBeforeHolidaysApply = JsonPath.from(loanDetails).get("repaymentSchedule");
+        final ArrayList<LinkedHashMap> periodsBeforeHolidaysApply = (ArrayList<LinkedHashMap>) repaymentScheduleHashMapBeforeHolidaysApply
+                .get("periods");
+
+        for (LinkedHashMap period : periodsBeforeHolidaysApply) {
+            final ArrayList<Integer> fromDate = (ArrayList<Integer>) period.get("fromDate");
+            if (fromDate != null) {
+                final Integer fromDateMonth = fromDate.get(1);
+                final Integer repaymentsRescheduledDateMonth = repaymentsRescheduledDate.get(1);
+                if (Objects.equals(fromDateMonth, repaymentsRescheduledDateMonth)) {
+                    final Integer repaymentsRescheduledDateDay = repaymentsRescheduledDate.get(2);
+                    final Integer fromDateDay = fromDate.get(2);
+                    Assertions.assertNotEquals(repaymentsRescheduledDateDay, fromDateDay,
+                            "Verifying Repayment Rescheduled Day before Running Apply Holidays to Loans Scheduler Job");
+                }
             }
         }
 
-        String JobName = "Apply Holidays To Loans";
+        String jobName = "Apply Holidays To Loans";
 
-        this.schedulerJobHelper.executeAndAwaitJob(JobName);
+        this.schedulerJobHelper.executeAndAwaitJob(jobName);
 
         // Loan Repayment Schedule After Apply Holidays To Loans
         loanDetails = this.loanTransactionHelper.getLoanDetails(requestSpec, responseSpec, loanID);
-        repaymentScheduleHashMap = JsonPath.from(loanDetails).get("repaymentSchedule");
-        periods = (ArrayList<LinkedHashMap>) repaymentScheduleHashMap.get("periods");
+        final LinkedHashMap repaymentScheduleHashMapAfterHolidaysApply = JsonPath.from(loanDetails).get("repaymentSchedule");
+        final ArrayList<LinkedHashMap> periodsAfterHolidaysApply = (ArrayList<LinkedHashMap>) repaymentScheduleHashMapAfterHolidaysApply
+                .get("periods");
         ArrayList<Integer> dateToApplyHolidays = null;
 
-        for (LinkedHashMap period : periods) {
-            ArrayList<Integer> fromDate = (ArrayList<Integer>) period.get("fromDate");
-            if (fromDate != null && Objects.equals(fromDate.get(1), repaymentsRescheduledDate.get(1))) {
-                dateToApplyHolidays = fromDate;
+        for (LinkedHashMap periodBeforeHolidaysApply : periodsAfterHolidaysApply) {
```

Comment:
Iterate twice on the same list? :o 
`for (LinkedHashMap periodBeforeHolidaysApply : periodsAfterHolidaysApply) {`
` for (LinkedHashMap periodAfterHolidaysApply : periodsAfterHolidaysApply) {`

Isnt there a more readable version to test this?

Reply:
Sorry, I fixed it, PTAL.